scipy/special/_support_cupy_torch_jax.py

```diff
@@ -0,0 +1,50 @@
+import sys
+from scipy._lib.array_api_compat.array_api_compat import array_namespace
+from . import _ufuncs
+from ._ufuncs import *
+
+__all__ = _ufuncs.__all__
+
+array_api_comat_prefix = "scipy._lib.array_api_compat.array_api_compat"
+
+
+def get_array_special_func(f_name, xp):
+    if xp.__name__ == f"{array_api_comat_prefix}.numpy":
```

Comment:
In https://github.com/scipy/scipy/pull/19005, I have added an `is_numpy` to `_array_api.py`, since this is required for every function which has an existing SciPy implementation (which doesn't just call `np`):
```python
def is_numpy(xp):
    return xp.__name__ == "scipy._lib.array_api_compat.array_api_compat.numpy"
```
`is_cupy` and `is_torch` are not used in the `fft` PR since we have not included any calls out to particular libraries like this PR does.

If work like this (calling out to these libraries) is going to become widespread across SciPy, then it certainly makes sense to have these helpers in `scipy._lib`.

Reply:
If gh-19005 merges first, I can incorporate that.